### PR TITLE
[#119] Smoother and more transparent transition to e consent

### PR DESCRIPTION
**Pull request by jhflorey**
This PR addresses issue https://github.com/episphere/connectApp/issues/119

- Add The next screens will take you through the informed consent process. They explain more about the study and what it means to take part in Connect. At the end, you can read the full informed consent form and electronic health records release form before you decide if you want to join. at the bottom of the "how did you hear about the study" question page near the submit/next button.

- Add "Previous" button on the Welcome Stage of the eConsent screen to allow the user to return to the How did you hear about the study page.

- Keep state of How did you hear about the study page while switching between How did you hear about the study page and eConsent screen.

### CODE REVIEW

**Review by anthonypetersen — COMMENTED** [review]
Code looks good, only thoughts are to not use hard-coded CIDs and instead reference a constant from the constants file. Thoughts? @we-ai

**Comment by we-ai** [comment]
> Code looks good, only thoughts are to not use hard-coded CIDs and instead reference a constant from the constants file. Thoughts? @we-ai
Agreed. We need to gradually replace hard-coded CIDs with more reader-friendly phrases/words/names.

**Review by we-ai — APPROVED** [review]

File: js/fieldToConceptIdMapping.js [hunk 1]
```diff
@@ -23,8 +23,33 @@ export default
     "healthcareProvider":827220437,
     "verification":821247024,
     "verified":197316935,
+    "yes": 353358909,
+    "no": 104430631,
     "consentVersion": 454205108,
     "hipaaVersion": 412000022,
+    "heardAboutStudyForm" : 142654897,
+
+    "heardAboutStudyList":{
```
Comment (we-ai) [inline js/fieldToConceptIdMapping.js:32]: This `List` is not a list (array). Will `checkBoxes` or `heardAboutStudyCheckBoxes` a better name?

File: js/event.js [hunk 2]
```diff
@@ -196,26 +197,26 @@ export const addEventHeardAboutStudy = () => {
...
+        formData[fieldMapping.heardAboutStudyForm][fieldMapping.heardAboutStudyList.checkbox14] = document.getElementById('checkbox14').checked ? fieldMapping.yes : fieldMapping.no;
+        formData[fieldMapping.heardAboutStudyForm][fieldMapping.heardAboutStudyList.checkbox15] = document.getElementById('checkbox15').checked ? fieldMapping.yes : fieldMapping.no;
+        formData[fieldMapping.heardAboutStudyForm][fieldMapping.heardAboutStudyList.checkbox16] = document.getElementById('checkbox16').checked ? fieldMapping.yes : fieldMapping.no;
+        formData[fieldMapping.heardAboutStudyForm][fieldMapping.heardAboutStudyList.checkbox17] = document.getElementById('checkbox17').checked ? fieldMapping.yes : fieldMapping.no;
+        formData[fieldMapping.heardAboutStudyForm][fieldMapping.heardAboutStudyList.checkbox18] = document.getElementById('checkbox18').checked ? fieldMapping.yes : fieldMapping.no;
+        formData[fieldMapping.heardAboutStudyForm][fieldMapping.heardAboutStudyList.checkbox19] = document.getElementById('checkbox19').checked ? fieldMapping.yes : fieldMapping.no;
```
Comment (we-ai) [inline js/event.js:220]: Above `formData[fieldMapping.heardAboutStudyForm]` is kind of long. We can use a variable to hold the data (eg `let inputData = {};`), and assign the data to `formData` after finishing updates (eg: `formData[fieldMapping.heardAboutStudyForm]= inputData;`)

Comment (anthonypetersen) [reply js/event.js:220]: Another idea for shortening things is to store the concepts in the constants file as such:

"heardAboutStudyList": {
checkbox1: 142654897.196856782,
checkbox2: 142654897.461488577,
etc...
}


Also one thing I'm thinking about now, should the names of those constants have a little more meaning than just "checkbox[number]"? I.E. the names could be "public event", or "familyOrFriend". Just a thought, don't feel too strongly one way or the other.

File: js/fieldToConceptIdMapping.js (hunk 1 above)
Comment (jhflorey) [reply js/fieldToConceptIdMapping.js]: Solved

File: js/event.js (hunk 2 above)
Comment (jhflorey) [reply js/event.js:226]: Solved

**Review by anthonypetersen — APPROVED** [review]
looks good

**Pull request merged**